### PR TITLE
CORE-2239+2242 OpenZeppelin ERC721 -> OpenZeppelin ERC721TokenReceiver

### PR DESCRIPTION
**Pull request by erer1243**
script that demonstrates ERC721 interactions between an OpenZeppelin based ERC721 and ERC721TokenReceiver

### CODE REVIEW

**Comment by erer1243** [comment]
i dont think those browser-test fails are my fault ?

File: examples/ERC721TokenReceiver/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.PHONY: build clean
+
+build:
+	test -d "openzeppelin-contracts-4.7.3" || tar -xf openzeppelin-contracts-4.7.3.tar.gz
+	mkdir -p build
+	solc --pretty-json --combined-json abi,bin oz_erc721.sol > build/oz_erc721.json
+	solc --pretty-json --combined-json abi,bin oz_erc721_tokenreceiver.sol > build/oz_erc721_tokenreceiver.json
```
Comment (jeapostrophe) [inline examples/ERC721TokenReceiver/Makefile:7]: You should not rely on solc in the build environment and you should not commit giant tar balls like this into the repo. You can customize the Dockerfile for the build, like this --- https://github.com/reach-sh/reach-lang/blob/master/examples/remote/Dockerfile

File: examples/ERC721TokenReceiver/index.mjs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    process.exit(0);
+}
+
+const acc = await stdlib.newTestAccount(stdlib.parseCurrency(100));
+
+const lock = () => {
+    let unlock;
+    const promise = new Promise(r => { unlock = r; });
+    return [unlock, promise];
+};
```
Comment (jeapostrophe) [inline examples/ERC721TokenReceiver/index.mjs:17]: This is basically `Signal` from the `stdlib`

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: examples/ERC721TokenReceiver/Makefile (hunk 1 above)
Comment (erer1243) [reply examples/ERC721TokenReceiver/Makefile:7]: well, in my defense, we have two other copies of the entire openzeppelin repo sitting around :p

Comment (jeapostrophe) [reply examples/ERC721TokenReceiver/Makefile:7]: No, we don't! They are downloaded as needed, not committed, just like I'm suggesting

File: examples/ERC721TokenReceiver/openzeppelin-contracts-4.7.3/LICENSE [hunk 3]
```diff
@@ -0,0 +1,22 @@
+The MIT License (MIT)
```
Comment (jeapostrophe) [inline examples/ERC721TokenReceiver/openzeppelin-contracts-4.7.3/LICENSE:1]: I feel like I must have been confusing before... DO NOT COMMIT openzeppelin into our repository in any way, whether as the raw files or as the tarball

**Review by jeapostrophe — CHANGES_REQUESTED** [review]

File: examples/ERC721TokenReceiver/openzeppelin-contracts-4.7.3/LICENSE (hunk 3 above)
Comment (erer1243) [reply examples/ERC721TokenReceiver/openzeppelin-contracts-4.7.3/LICENSE:1]: Ok. I was going off of the Jira ticket description of using a "local copy of openzeppelin

Comment (jeapostrophe) [reply examples/ERC721TokenReceiver/openzeppelin-contracts-4.7.3/LICENSE]: I'm sorry for being vague.

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**